dapr/clients/grpc/dapr_client.py

```diff
@@ -0,0 +1,168 @@
+# -*- coding: utf-8 -*-
+
+"""
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT License.
+"""
+
+import grpc  # type: ignore
+
+from dapr.conf import settings
+from dapr.clients.base import DEFAULT_JSON_CONTENT_TYPE
+
+from google.protobuf.any_pb2 import Any as GrpcAny
+from google.protobuf.message import Message as GrpcMessage
+
+from dapr.proto import api_v1, api_service_v1, common_v1
+from typing import Dict, List, Optional, Union, Tuple
+
+
+MetadataDict = Dict[str, List[Union[bytes, str]]]
+MetadataTuple = Tuple[Tuple[str, Union[bytes, str]], ...]
+
+
+class InvokeServiceRequestData:
+    def __init__(
+            self,
+            data: Union[bytes, GrpcMessage],
+            content_type: Optional[str] = None):
+        self._data = GrpcAny()
+        if isinstance(data, bytes):
+            self._data.value = data
+            self._content_type = content_type
+            if not content_type:
+                self._content_type = DEFAULT_JSON_CONTENT_TYPE
+        elif isinstance(data, GrpcMessage):
+            self._data.Pack(data)
+            self._content_type = None
+        else:
+            raise ValueError(f'invalid data type {type(data)}')
+
+    @property
+    def proto(self) -> GrpcAny:
+        return self._data
+
+    @property
+    def content_type(self) -> Optional[str]:
+        return self._content_type
+
+
+class DaprResponse:
```

Comment:
@shalabhms I have thought about how we can retrieve trace id easily. my initial idea is to have `DaprResponse` object which allows users to get any headers. trace + span id will be one of headers.. so that they can use it.

But. we need to discuss it more.

Reply:
@youngbupark , yes thats the right direction. I was having similar discussion with @amanbha , as how to do it in dot net sdk. we can discuss it more on this also we have to think a way/expose a helper that user can pass any metadata.